### PR TITLE
Rate bugfix done, now the user has an array of the books that he rated a...

### PR DESCRIPTION
**Pull request by KarimElSheikh**
...nd his rating values for each book. Also, the book now stores the average and the sum of ratings and keeps them updated #82
@Vegas @Mina @Israa @omarkamal95 @Mernawaseif

### CODE REVIEW

File: backend/views/book.js [hunk 1]
```diff
@@ -286,24 +304,54 @@ module.exports = function(router) {
...
 			else {
+				book.ratingsSum += rating;
 				book.markModified('ratings');
+				book.avgRating = book.ratingsSum / book.ratings.length;
 				book.save();
-				res.json("Updated the book rating successfully");
 			}
 
-		})
+			User.findOne({'_id': userId}, function findBookText(err, user) {
```
Comment (MinaNemr) [inline backend/views/book.js:327]: @KarimElSheikh 
this function name is wrong
it is not findBookText
please either remove the name or name it with another name and document it

**Comment by MinaNemr** [comment]
@KarimElSheikh u need to fix something i mentioned on a note on book.js

**Comment by Veegas** [comment]
@KarimElSheikh Sheel folder ./idea, w e3ml el ratings el fel user of type ObjectID

**Comment by KarimElSheikh** [comment]
@MinaNemr I removed the function name, the function that call this inner function is named and documented
@Veegas I removed the .idea folder. About changing the type to ObjectID, you can always convert from the string to ObjectID using "var x = mongoose.Types.ObjectId([String])"
I didn't do it because the function Books.findOne({'_id': bookId} doesn't work if bookId is an ObjectID instead of string

**Comment by Veegas** [comment]
@KarimElSheikh  La2 msh haynfa3 tesebha of type Mixed kda 3ashan lama hateegy te-view el books that the user rated lama teegy tegeeb el record msh hate3raf te3mlo populate

**Comment by Israa-Elwan** [comment]
@KarimElSheikh  ur scenario are correct in your branch ! but u need to solve the conflicts in order to test the scenarios after solving them

**Comment by KarimElSheikh** [comment]
@Veegas momken te3mel function teconvert :D

**Pull request merged**